homeassistant/components/downloader/config_flow.py

```diff
@@ -55,8 +55,9 @@ async def async_step_import(
 
     async def _validate_input(self, user_input: dict[str, Any]) -> None:
         """Validate the user input if the directory exists."""
-        if not os.path.isabs(user_input[CONF_DOWNLOAD_DIR]):
-            download_path = self.hass.config.path(user_input[CONF_DOWNLOAD_DIR])
+        download_path = user_input[CONF_DOWNLOAD_DIR]
+        if not os.path.isabs(download_path):
+            download_path = self.hass.config.path(download_path)
 
         if not os.path.isdir(download_path):
```

Comment:
Side note: `os.path.isdir` should be run on the executor thread pool to avoid blocking the event loop.